### PR TITLE
fix(select): allow for null values in HTML select options bound with ngValue

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
supersedes/closes #12026

### CODE REVIEW

**Comment by vicb** [comment]
what's the original PR # ?

**Comment by DzmitryShylovich** [comment]
https://github.com/angular/angular/pull/12026

File: modules/@angular/forms/src/directives/select_multiple_control_value_accessor.ts [hunk 1]
```diff
@@ -6,13 +6,11 @@
...
 
-import {Directive, ElementRef, Host, Input, OnDestroy, OpaqueToken, Optional, Renderer, Type, forwardRef} from '@angular/core';
-
+import {Directive, ElementRef, Host, Input, OnDestroy, Optional, Renderer, forwardRef} from '@angular/core';
 import {isPrimitive, looseIdentical} from '../facade/lang';
-
 import {ControlValueAccessor, NG_VALUE_ACCESSOR} from './control_value_accessor';
 
-export const SELECT_MULTIPLE_VALUE_ACCESSOR = {
+export const SELECT_MULTIPLE_VALUE_ACCESSOR: any = {
```
Comment (vicb) [inline modules/@angular/forms/src/directives/select_multiple_control_value_accessor.ts]: any -> Provider

File: modules/@angular/forms/test/template_integration_spec.ts [hunk 2]
```diff
@@ -699,6 +699,30 @@ export function main() {
            expect(select.nativeElement.value).toEqual('2: Object');
            expect(secondNYC.nativeElement.selected).toBe(true);
          }));
+
+      it('should work with null option', fakeAsync(() => {
+           const fixture = TestBed.createComponent(NgModelSelectWithNullForm);
+           const comp = fixture.componentInstance;
+           comp.cities = [{'name': 'SF'}, {'name': 'NYC'}, {'name': 'NYC'}];
```
Comment (vicb) [inline modules/@angular/forms/test/template_integration_spec.ts]: remove the duplicate NYC here probably

**Comment by vicb** [comment]
LGTM with a few nits.

@kara could you please review and add the "merge" label ? (you LGTM'ed the original PR, this one adds a test)

File: modules/@angular/forms/test/template_integration_spec.ts [hunk 3]
```diff
@@ -699,6 +699,28 @@ export function main() {
...
+      it('should work with null option', fakeAsync(() => {
+           const fixture = TestBed.createComponent(NgModelSelectWithNullForm);
+           const comp = fixture.componentInstance;
+           comp.cities = [{'name': 'SF'}, {'name': 'NYC'}];
+           comp.selectedCity = null;
+           fixture.detectChanges();
+
+           const select = fixture.debugElement.query(By.css('select'));
+
+           select.nativeElement.value = '2: Object';
```
Comment (DzmitryShylovich) [inline modules/@angular/forms/test/template_integration_spec.ts]: `<option [ngValue]="null">Unspecified</option>` === `'0: null'`
`{'name': 'SF'}` === `'1: Object'`
`{'name': 'NYC'}` === `'2: Object'`

Comment (vicb) [inline modules/@angular/forms/test/template_integration_spec.ts]: yes, you're right

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**